astronomer/providers/amazon/aws/hooks/batch_client.py

```diff
@@ -0,0 +1,256 @@
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.batch_client import BatchClientHook
+
+from astronomer.providers.amazon.aws.hooks.base_aws_async import AwsBaseHookAsync
+
+
+class BatchClientHookAsync(BatchClientHook, AwsBaseHookAsync):
+    """
+    Async client for AWS Batch services.
+
+    :param max_retries: exponential back-off retries, 4200 = 48 hours;
+        polling is only used when waiters is None
+
+    :param status_retries: number of HTTP retries to get job status, 10;
+        polling is only used when waiters is None
+
+    .. note::
+        Several methods use a default random delay to check or poll for job status, i.e.
+        ``random.sample()``
+        Using a random interval helps to avoid AWS API throttle limits
+        when many concurrent tasks request job-descriptions.
+
+        To modify the global defaults for the range of jitter allowed when a
+        random delay is used to check Batch job status, modify these defaults, e.g.:
+        .. code-block::
+
+            BatchClient.DEFAULT_DELAY_MIN = 0
+            BatchClient.DEFAULT_DELAY_MAX = 5
+
+        When explicit delay values are used, a 1 second random jitter is applied to the
+        delay .  It is generally recommended that random jitter is added to API requests.
+        A convenience method is provided for this, e.g. to get a random delay of
+        10 sec +/- 5 sec: ``delay = BatchClient.add_jitter(10, width=5, minima=0)``
+    .. see also::
+        - https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/batch.html
+        - https://docs.aws.amazon.com/general/latest/gr/api-retries.html
+        - https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
+    """
+
+    def __init__(self, job_id: Optional[str], waiters: Any, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.job_id = job_id
+        self.waiters = waiters
+
+    async def monitor_job(self) -> Union[Dict[str, str], None]:
+        """
+        Monitor an AWS Batch job
+        monitor_job can raise an exception or an AirflowTaskTimeout can be raised if execution_timeout
+        is given while creating the task. These exceptions should be handled in taskinstance.py
+        instead of here like it was previously done
+
+        :raises: AirflowException
+        """
+        if not self.job_id:
+            raise AirflowException("AWS Batch job - job_id was not found")
+
+        if self.waiters:
+            self.waiters.wait_for_job(self.job_id)
+            return None
+        else:
+            await self.wait_for_job(self.job_id)
+            await self.check_job_success(self.job_id)
+            success_msg = "AWS Batch job (%s) succeeded" % self.job_id
+            self.log.info(success_msg)
+            return {"status": "success", "message": success_msg}
+
+    async def check_job_success(self, job_id: str) -> bool:  # type: ignore[override]
```

Comment:
why do we need ```# type: ignore[override]```  here?

Reply:
because mypy will give the below error, and this `ignore[override]` tag present all across our repository

 `error: Return type "Coroutine[Any, Any, bool]" of "check_job_success" incompatible with return type "bool" in supertype "BatchClientHook"  [override]`